Sync/SyncStateMachine.swift

```diff
@@ -40,22 +40,40 @@ private let DefaultEngines: [String: Int] = [
 // meta/global produced locally.
 private let DefaultDeclined: [String] = [String]()
 
-// public for testing.
-public func createMetaGlobalWithEngineConfiguration(_ engineConfiguration: EngineConfiguration) -> MetaGlobal {
+public func computeNewEngines(_ engineConfiguration: EngineConfiguration, enginesEnablements: [String: Bool]?) -> (engines: [String: EngineMeta], declined: [String]) {
+    var enabled: Set<String> = Set(engineConfiguration.enabled)
+    var declined: Set<String> = Set(engineConfiguration.declined)
     var engines: [String: EngineMeta] = [:]
-    for engine in engineConfiguration.enabled {
+
+    if let enginesEnablements = enginesEnablements {
+        let enabledLocally = Set(enginesEnablements.filter { $0.value }.map { $0.key })
+        let declinedLocally = Set(enginesEnablements.filter { !$0.value }.map { $0.key })
```

Comment:
One wonders if it wouldn't be better to have the enablements be two sets, rather than a map.

Reply:
I started with that, then switched back to a single dictionary because of implementations reasons that are not there anymore.
I wouldn't mind switching back, but I believe this is probably good enough.